ElectronicObserver/Window/Wpf/ShipTrainingPlanner/ShipTrainingPlanViewerViewModel.cs

```diff
@@ -98,7 +100,10 @@ private void Initialize(string apiname, dynamic data)
 
 	private void OnPlanViewModelPropertyChanged(object? sender, PropertyChangedEventArgs e)
 	{
-		if (e.PropertyName is nameof(ShipTrainingPlanViewModel.PlanFinished)) UpdatePlanList();
+		if (e.PropertyName is not nameof(ShipTrainingPlanViewModel.PlanFinished)) return;
+
+		UpdatePlanList();
+		if (sender is ShipTrainingPlanViewModel { PlanFinished: true } plan) OnPlanCompleted?.Invoke(plan);
```

Comment:
Single line ifs should only be for things like return, break, continue, throw etc.
Prefer a normal multi line if when there's some code happening.